components/global/core/io.cellery.observability.k8s.client/src/main/java/io/cellery/observability/k8s/client/ComponentsEventSource.java

```diff
@@ -0,0 +1,264 @@
+                        attributes.put(Constants.ATTRIBUTE_COMPONENT, componentName);
+                        attributes.put(INGRESS_TYPES,  StringUtils.join(ingressTypesList, ','));
+                        sourceEventListener.onEvent(attributes, new String[0]);
+                    }
+
+
+                } catch (ParseException e) {
+                    logger.error("Ignored cell change due to creation timestamp parse failure", e);
+                }
+            }
+
+            @Override
+            public void onClose(KubernetesClientException cause) {
+                if (cause == null) {
+                    if (logger.isDebugEnabled()) {
+                        logger.debug("Kubernetes cell watcher closed successfully");
+                    }
+                } else {
+                    logger.error("Kubernetes cell watcher closed with error", cause);
+                }
+            }
+        });
+        if (logger.isDebugEnabled()) {
+            logger.debug("Created cell watcher");
+        }
+    }
+
+    @Override
+    public void disconnect() {
+        if (cellWatcher != null) {
+            cellWatcher.close();
+            if (logger.isDebugEnabled()) {
+                logger.debug("Closed cell watcher");
+            }
+        }
+        if (k8sClient != null) {
+            k8sClient.close();
+            if (logger.isDebugEnabled()) {
+                logger.debug("Closed API server client");
+            }
+        }
+    }
+
+    @Override
+    public void destroy() {
+        // Do nothing
+    }
+
+    @Override
+    public void pause() {
+        // Do nothing
+    }
+
+    @Override
+    public void resume() {
+        // Do nothing
+    }
+
+    @Override
+    public Map<String, Object> currentState() {
+        return null;
+    }
+
+    @Override
+    public void restoreState(Map<String, Object> map) {
+        // Do nothing
+    }
+
+    /**
+     * This method checks if the ingress type is of HTTP and adds it to the relevant cell data
+     */
+    private void addHttpType(List<HTTP> httpObjectsList, String componentName,
+                             List<String> ingressTypes, boolean isWebCell) {
+        if (httpObjectsList != null) {
+            boolean isHttp = httpObjectsList.stream().anyMatch(httpObj -> httpObj.getBackend()
+                    .equals(componentName));
+            // Check for Web ingresses
+            if (isHttp && isWebCell) {
+                ingressTypes.add(Constants.INGRESS_TYPE_WEB);
+            } else if (isHttp) {
+                ingressTypes.add(Constants.INGRESS_TYPE_HTTP);
+            }
+        }
+    }
+
+    /**
+     * This method checks if the ingress type is of TCP type and adds it to the relevant cell data
+     */
+    private void addTcpType(List<TCP> tcpObjectsList, String componentName, List<String> ingressTypes) {
+        if (tcpObjectsList != null) {
+            boolean isTCP = tcpObjectsList.stream().anyMatch(tcpObject -> tcpObject.getBackendHost()
```

Comment:
You don't need to assign a variable to this check, you can move the if condition clause up.